File: P171 Excel Sheet Column Number/Solution.c
int titleToNumber(char* s) {
     if (s[0]=='\0' || strlen(s) == 0 ) return 0;
        int result = 0;
        for(int i=0;i<strlen(s);i++)
        {
            result = result + (s[i]-'A'+1) * (int) pow(26, strlen(s)-1-i);  
        }
        return result;
}

File: P171 Excel Sheet Column Number/Solution.java
public class Solution {
    public int titleToNumber(String s) {
        int num = 0;
        int time = 1;
        for (int i = s.length() - 1; i >= 0; i--) {
            num += time * (s.charAt(i) - 64);
            time *= 26;
        }
        return num;
    }

    public int titleToNumber2(String s) {
       if (s == null || s.length()==0) return 0;
       HashMap<Character, Integer> hashMap = new HashMap<Character, Integer>();
       char c = 'A';
       for(int i=1; i<=26; i++){
           hashMap.put(c, i);
           c += 1; 
       }
       int result = 0;
       for(int i=0;i<s.length();i++)
       {
           result = result + hashMap.get(s.charAt(i)) * (int) Math.pow(26, s.length()-1-i); 
       }
       return result;
    }
}